playpen/all_27.sh

```diff
@@ -1,22 +1,52 @@
 #!/bin/bash
+# Downgrade a Pulp system from master to 2.7.
+#
+# WARNING: This script drops a mongo database, uninstalls software, and
+# performs other destructive actions.
+source ~/.bashrc  # get pstop, pstart, etc
+set -euo pipefail
 
-#  When using this script, you will need to fix your database accordingly. If moving backward, you
-#  will need to drop the database, and you will always need to run the migrations.
+pstop
 
-
-# pulp_docker, pulp_python, pulp_ostree excluded because they are versioned differently
-for repo in pulp pulp_puppet pulp_rpm
-do
-    pushd /home/vagrant/devel/$repo
-    git checkout 2.7-dev
-    find ~/devel -name "*.py[c0]" -delete
-    sudo ./pulp-dev.py --install
-    popd
+# Downgrade these plugins to version 2.7.
+pushd ~/devel
+for repo in pulp{,_puppet,_rpm}; do
+    if [ -d "$repo" ]; then
+        pushd "$repo"
+        git checkout 2.7-dev
+        find . -name '*.py[c0]' -delete
+        sudo ./pulp-dev.py --install
+        popd
+    fi
 done
+popd
 
-for repo in pulp_docker pulp_python
-do
-    pushd /home/vagrant/devel/$repo
-    sudo ./pulp-dev.py --uninstall
-    popd
+# Completely uninstall these plugins. (Why?)
```

Comment:
pulp_python and pulp_docker are versioned independently from pulp and the other plugins so they cannot be installed with the loop above which uses `2.7-dev`. When I wrote this script, I wasn't using either of these plugins so I just left them uninstalled. Please feel free to add something to check out the appropriate version for each and install it.